src/fs_clonefile.cpp

```diff
@@ -67,24 +67,24 @@ writen(const int     fd,
           return -1;
         }
 
-      nleft -= nwritten;
-      buf   += nwritten;
+      nleft -= (size_t)nwritten;
+      buf   += (size_t)nwritten;
     }
 
-  return count;
+  return nwritten;
```

Comment:
I get that for the outer function (copyfile_rw) but for writen the only possible bug should be that it doesn't accept zero for byte length. writen should not return till all bytes passed in are written or an error.

I think the issue is actually in copyfile_rw in which a zero return from read isn't handled properly. Should exit the loop.